sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableUtils.java

```diff
@@ -159,6 +168,27 @@ public static <E extends HttpResponseException> Mono<Response<Void>> swallowExce
         return monoError(logger, httpResponseException);
     }
 
+    public static Context setContext(Context context) {
+        return setContext(context, false);
+    }
+
+    public static Context setContext(Context context, boolean isSync) {
+        Context val = context != null ? context : Context.NONE;
+        return isSync ? enableSyncRestProxy(setTrailingContext(val)) : setTrailingContext(val);
+    }
+
+    private static Context setTrailingContext(Context context) {
+        return context.addData(AZ_TRACING_NAMESPACE_KEY, TABLES_TRACING_NAMESPACE_VALUE);
+    }
+
+    private static Context enableSyncRestProxy(Context context) {
+        return context.addData(HTTP_REST_PROXY_SYNC_PROXY_ENABLE, true);
+    }
+
+    public static Long setTimeout(Duration timeout) {
+        return timeout != null ? timeout.toMillis() : Duration.ofDays(1).toMillis();
```

Comment:
What would this timeout be normally used for? One day seems like a really long time to be a default for things like an operation timeout, for example.

Reply:
@vcolin7 This is an additional timeout parameter that the user can set since the sync methods take a timeout parameter.  The timeout is not placed onto the request and so it just once the request fails (for a pipeline timeout, bad status code, etc.) then it returns an exception which exits the wait period. The thing to be aware of is that the default for the timeout needs to be longer than the collective amount of waiting that the pipeline will allot a request, which is why I chose one day.